### PR TITLE
feat: Add circom-lsp

### PR DESCRIPTION
**Pull request by Gokul2003g**
Addresses the feature request #2833 
Added circom-lsp.lua.
Ran all the linting checks. 

Please look into the PR and merge if OK.

Thanks

### CODE REVIEW

File: lua/lspconfig/server_configurations/circom-lsp.lua [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  'pyproject.toml',
+  'Cargo.toml',
+}
+
+return {
+  default_config = {
+    cmd = { 'circom-lsp' },
+    filetypes = { 'circom' },
+    root_dir = function()
+      local root = util.root_pattern(root_files)(vim.fn.expand '%:p')
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/circom-lsp.lua:15]: there no need `expand` just `fname`

File: lua/lspconfig/server_configurations/circom-lsp.lua [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      return root or vim.api.nvim_buf_get_name(0)
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/circom-lsp.lua:16]: i don't think use current bufname is useful if you want root is current dir just `single_file_support = true`

Comment (Gokul2003g) [reply lua/lspconfig/server_configurations/circom-lsp.lua:16]: Yes I agree 
Simplify the configuration by setting single_file_support to true. This approach is more straightforward, as you don't need to specify root_files or create a custom root_dir function.

Comment (Gokul2003g) [reply lua/lspconfig/server_configurations/circom-lsp.lua]: Set single_file_support to true for improved simplicity.

**Comment by Gokul2003g** [comment]
Hey @glepnir Can you please review the PR whenever you are free?

File: lua/lspconfig/server_configurations/circom-lsp.lua [hunk 3]
```diff
@@ -0,0 +1,14 @@
+return {
+  default_config = {
+    cmd = { 'circom-lsp' },
+    filetypes = { 'circom' },
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/circom-lsp.lua:4]: `    root_dir = util.find_git_ancestor,`

Comment (Gokul2003g) [reply lua/lspconfig/server_configurations/circom-lsp.lua:4]: adding  `root_dir = util.find_git_ancestor,`

Comment (Gokul2003g) [reply lua/lspconfig/server_configurations/circom-lsp.lua:6]: - The root_dir is now set to util.find_git_ancestor, allowing the language server to determine the project's root directory based on Git repositories. This enhances project context when working within Git-managed projects.

- single_file_support is enabled with a value of true. This setting allows the language server to operate in the current directory when working with single Circom files, providing improved support for single-file projects.

**Pull request merged**